asv_bench/benchmarks/series_methods.py

```diff
@@ -90,6 +90,18 @@ def time_isin_long_series_long_values_floats(self):
         self.s_long_floats.isin(self.vals_long_floats)
 
 
+class IsInLongSeries(object):
+    params = [["int64", "int32", "float64", "float32"], [1, 2, 5, 10, 1000, 10 ** 5]]
+    param_names = ["dtype", "M"]
+
+    def setup(self, dtype, M):
+        self.s = Series(np.random.randint(0, M, 10 ** 7)).astype(dtype)
```

Comment:
IIUC we can expect big perf differences if the values here are clustered, e.g. if they are all negative or all >M in the extreme cases.  Do we need to measure any non-uniform cases?

Reply:
@jbrockmendel You are right, the test is a little bit naive. We should test at least (I have pushed them):

   - average (random) case with elements in the look-up-table
   - average (random) case with elements not in the look-up-table
   - monotone series with elements in the look-up-table (better cache utilization for numpy's approach)
   - monotone series with elements not in the look-up-table (better cache utilization for numpy's approach)

it looks as if float32/float64 are somewhat problematic, hopefully after #36729 it will become clearer what is going on.